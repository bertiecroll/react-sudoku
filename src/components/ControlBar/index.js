export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
