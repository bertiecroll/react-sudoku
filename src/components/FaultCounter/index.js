export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
