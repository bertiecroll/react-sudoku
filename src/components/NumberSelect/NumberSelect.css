.NumberSelect {
  margin: 1em;
  padding: .25em;
  display: flex;
  flex-direction: row;
}
