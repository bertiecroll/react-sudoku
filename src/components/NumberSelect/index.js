export { NumberSelect } from './NumberSelect'
export { default } from './NumberSelect'
