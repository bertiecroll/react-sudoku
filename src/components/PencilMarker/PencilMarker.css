.PencilMarker .NumberSelect .NumberButton {
  background-color: #fff;
  color: #67DAF9;
}

.PencilMarker .NumberSelect .not-available {
  color: white;
}
