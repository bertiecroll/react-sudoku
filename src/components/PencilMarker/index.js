export { PencilMarker } from './PencilMarker'
export { default } from './PencilMarker'
