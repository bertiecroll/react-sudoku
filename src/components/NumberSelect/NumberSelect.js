import React from 'react'
import PropTypes from 'prop-types'
import './NumberSelect.css'

export const NumberSelect = (props) => {
  return (
    <div className="NumberSelect">
      {props.availableNumbers.map(number => <div key={number} onClick={() => props.handleClick(number)}>{number}</div>)}
    </div>
  )
}

NumberSelect.propTypes = {
  availableNumbers: PropTypes.array.isRequired
}
