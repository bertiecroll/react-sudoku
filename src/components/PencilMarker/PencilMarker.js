import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import './PencilMarker.css'
import NumberSelect from 'components/NumberSelect'
import { addPencilMark, removePencilMark } from 'redux/actions/grid'

export function PencilMarker ({ addPencilMark, removePencilMark }) {

  const onClickAction = (selectedCell, selectedNumber) => {
    const { id, pencilMarks } = selectedCell
    pencilMarks.includes(selectedNumber) ? removePencilMark(id, selectedNumber) : addPencilMark(id, selectedNumber)
  }

  return (
    <div className="PencilMarker">
      <NumberSelect onClickAction={onClickAction} />
    </div>
  )
}

const mapDispatchToProps = { addPencilMark, removePencilMark }
export default connect(null, mapDispatchToProps)(PencilMarker)

PencilMarker.propTypes = {
  addPencilMark: PropTypes.func,
  removePencilMark: PropTypes.func,
}
