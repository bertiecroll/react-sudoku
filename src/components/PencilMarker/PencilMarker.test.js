import React from 'react'
import { shallow } from 'enzyme'
import { PencilMarker } from 'components/PencilMarker'
import NumberSelect from 'components/NumberSelect'

describe('PencilMarker', () => {
  let wrapper
  // our mock redux action functions
  const mockAddPencilMark = jest.fn()
  const mockRemovePencilMark = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <PencilMarker
        addPencilMark={mockAddPencilMark}
        removePencilMark={mockRemovePencilMark}
      />
    )
  })

  afterEach(() => {
    mockAddPencilMark.mockClear()
    mockRemovePencilMark.mockClear()
  })

  it('renders NumberSelect component passing onClickAction', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('#onClickAction', () => {
    const selectedCell = { id: '0,0', pencilMarks: [1, 2, 3] }

    it('calls #addPencilMark when selectedCell pencilMarks does not include selectedNuber', () => {
      const selectedNumber = 4

      wrapper.find(NumberSelect).props().onClickAction(selectedCell, selectedNumber)

      expect(mockAddPencilMark).toHaveBeenCalledWith(selectedCell.id, selectedNumber)
    })

    it('calls #removePencilMark when selectedCell pencilMarks includes selectedNuber', () => {
      const selectedNumber = 1

      wrapper.find(NumberSelect).props().onClickAction(selectedCell, selectedNumber)

      expect(mockRemovePencilMark).toHaveBeenCalledWith(selectedCell.id, selectedNumber)
    })
  })
})
