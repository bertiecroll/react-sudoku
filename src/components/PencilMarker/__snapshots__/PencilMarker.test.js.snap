// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PencilMarker renders NumberSelect component passing onClickAction 1`] = `
<div
  className="PencilMarker"
>
  <Connect(NumberSelect)
    onClickAction={[Function]}
  />
</div>
`;
